### PR TITLE
Adds matchers for java.sql.Timestamp

### PR DESCRIPTION
**Pull request by ashishkujoy**
Matchers added so far

1. shouldBeBefore
2. shouldNotBeBefore
3. shouldBeAfter
4. shouldNotBeAfter

### CODE REVIEW

File: kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import io.kotest.MatcherResult
+import io.kotest.should
+import io.kotest.shouldNot
+import java.sql.Timestamp
+
+fun sameTimestamp(timestamp: Timestamp) = object: Matcher<Timestamp> {
+   override fun test(value: Timestamp): MatcherResult {
+      return MatcherResult(
+         timestamp.equals(value),
+         {"Expected $timestamp to be equal $value, but its not."},
```
Comment (LeoColman) [inline kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: nit:
```suggestion
         {"Expected $timestamp to be equal $value, but it's not."},
```

File: kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+         {"$timestamp expected to be equal $value, but it is equal"}
+      )
+   }
+}
+
+fun afterTimestamp(timestamp: Timestamp) = object: Matcher<Timestamp>  {
+   override fun test(value: Timestamp): MatcherResult {
+      return MatcherResult(
+         value.after(timestamp),
+         {"Expected $timestamp to be after $value, but its not."},
```
Comment (LeoColman) [inline kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: nit: 
```suggestion
         {"Expected $timestamp to be after $value, but it's not."},
```

**Comment by sksamuel** [comment]
Quick work @ashishkujoy

**Comment by LeoColman** [comment]
@ashishkujoy 
Would you mind creating a new PR updating the documentation as well?
This one is great!

**Review by LeoColman — APPROVED** [review]

**Comment by ashishkujoy** [comment]
> @ashishkujoy
> Would you mind creating a new PR updating the documentation as well?
> This one is great!

Sure

File: kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      )
+   }
+}
+
+
+/**
+ * Assert that [Timestamp] is equal to [anotherTimestamp].
+ * @see [shouldNotBe]
+ * */
+infix fun Timestamp.shouldBe(anotherTimestamp: Timestamp) = this should sameTimestamp(anotherTimestamp)
```
Comment (LeoColman) [inline kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: Ok, on a further review, I believe we don't want a `shouldBe` and a `shouldNotBe` for this guy.

We already have the default `shouldBe` matcher that compares using `==`, so this is redundant and unnecessary.

Could you please take them out of this commit and from the documentation commit?

**Review by LeoColman — CHANGES_REQUESTED** [review]

File: kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt (hunk 3 above)
Comment (ashishkujoy) [reply kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: One thing here is error message which is specific to timestamp when use Timestamp.shouldBe while in the regular one error message is generic. If we don't want specific error then i can surely remove shouldBe and shouldNotBe from Timestamp as well as from Instant add under [PR](https://github.com/kotlintest/kotlintest/issues/1127)

Comment (LeoColman) [reply kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: The default should be message is `expected: $expected but was: $actual`
I don't think the message difference has enough relevance to justify this change

What do you think?

Comment (ashishkujoy) [reply kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: Yes now i also think we don't need shouldBe and shouldNotBe i will remove those methods.

Comment (ashishkujoy) [reply kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/date/timestamp.kt]: Done

File: kotest-assertions/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/timestampTest.kt [hunk 4]
```diff
@@ -7,17 +7,6 @@ import java.time.Instant
...
-         val nowInstance = Instant.now()
-         Timestamp.from(nowInstance) shouldBe Timestamp.from(nowInstance)
-      }
-
-      "two timestamp of different instance should be not be same" {
-         val nowInstance = Instant.now()
-         val anotherInstance = nowInstance.plusMillis(1000L)
-         Timestamp.from(nowInstance) shouldNotBe Timestamp.from(anotherInstance)
-      }
-
```
Comment (LeoColman) [inline kotest-assertions/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/timestampTest.kt]: I think we can keep these tests

Comment (ashishkujoy) [reply kotest-assertions/src/jvmTest/kotlin/com/sksamuel/kotest/matchers/timestampTest.kt]: Done

**Comment by sksamuel** [comment]
Is this good for merge @kerooker

**Comment by LeoColman** [comment]
I think it is, yes

**Review by LeoColman — APPROVED** [review]

**Pull request merged**